synapse/storage/profile.py

```diff
@@ -62,6 +65,61 @@ def get_profile_avatar_url(self, user_localpart):
             desc="get_profile_avatar_url",
         )
 
+    @defer.inlineCallbacks
+    def get_latest_profile_replication_batch_number(self):
+        def f(txn):
+            txn.execute("SELECT MAX(batch) as maxbatch FROM profiles")
+            rows = self.cursor_to_dict(txn)
+            return rows[0]['maxbatch']
+        max_batch = yield self.runInteraction(
```

Comment:
you could just `return self.runInteraction(...)` rather than using `inlineCallbacks`

Reply:
likewise below.

Reply:
done